### ISSUE TITLE
Missing modulePyrapose.utils.compute_overlap

### ISSUE THREAD

**Issue by DRM-Free** [sThalham/RGBDPose#1]
While training I get this error :
Traceback (most recent call last):
  File "./train.py", line 35, in <module>
    from .. import layers  # noqa: F401
  File "./../../PyraPose/layers/__init__.py", line 1, in <module>
    from ._misc import RegressBoxes, RegressBoxes3D, DenormBoxes3D, UpsampleLike, Anchors, ClipBoxes  # noqa: F401
  File "./../../PyraPose/layers/_misc.py", line 19, in <module>
    from ..utils import anchors as utils_anchors
  File "./../../PyraPose/utils/anchors.py", line 23, in <module>
    from ..utils.compute_overlap import compute_overlap
ModuleNotFoundError: No module named 'PyraPose.utils.compute_overlap'

**Comment by sThalham** [comment]
Executing "python setup.py build_ext --inplace" in base directory builds the file.

**Comment by DRM-Free** [comment]
Hello and thank you for the swift response.
I just retried executing the setup file, to no avail.
Inside my utils folder I found related files :
compute_overlap.c  compute_overlap.pyx  compute_overlap.py.safe
compute_overlap.cpython-39-x86_64-linux-gnu.so
But no compute_overlap.py.

**Comment by sThalham** [comment]
What's the output of ""python setup.py build_ext --inplace"?
Are you executing it in the parent directory.

You shouldn't need the .py-file. There's a .py.safe. You could use that if linking the C-code is not working. 
The relevant file looks something like that "compute_overlap.cpython-36m-x86_64-linux-gnu.so". It's built after executing the mentioned command.

**Comment by DRM-Free** [comment]
Here is the output :
% python setup.py build_ext --inplace
running build_ext
skipping 'PyraPose/utils/compute_overlap.c' Cython extension (up-to-date)
copying build/lib.linux-x86_64-3.9/PyraPose/utils/
compute_overlap.cpython-39-x86_64-linux-gnu.so -> PyraPose/utils

**Comment by sThalham** [comment]
Sorry, I have no idea why it's not working for you. 

Maybe try to remove 'compute_overlap.c' and 'compute_overlap.cpython-36m-x86_64-linux-gnu.so' from utils and invoke setup.py again. Maybe that resolves it. 

Are you using Windows?

**Comment by DRM-Free** [comment]
Thanks for trying to help.
I should probably try to run Pyrapose outside of Docker for now as I have
little knowledge of how it works and it makes debugging less convenient. Do
you think it is possible without much trouble if I install the dependencies
? As much as I can tell,modifying source files does not affect the
execution of my docker images. Should I rebuild my docker image after
removing 'compute_overlap.c' and '
compute_overlap.cpython-36m-x86_64-linux-gnu.so' from utils ?
No, I am not using Windows but archlinux.

**Comment by sThalham** [comment]
It's running fine outside docker by installing the dependencies. But, yet I don't know where your problem with 'compute_overlap' is coming from. I regularly clone and compile in new containers, using the provided Dockerfile. And I have no problems whatsoever.

But as I said, you can switch to 'compute_overlap.py.safe', just remove the save at the end, the .c- and the .so-file and do the overlap computation in python. It will be slower though.

I'm not exactly sure what you're asking regarding docker.
But, no, modifying the running container does not change the image. To update the image with the changes in your container you have to commit the changes create a new image from it.

**Comment by DRM-Free** [comment]
Hello again,

I have news regarding my docker issues. I just checked and it looks like
the file 'compute_overlap.c' is indeed created in the utils directory when
running the setup script. However, when inspecting my generated docker
image, this file is missing. I have no idea how this is possible.

**Comment by sThalham** [comment]
That sounds like a docker issue to me. When do you build 'compute_overlap.c'? During image creation or manually in the running container. I usually invoke the command manually in the container.

**Comment by DRM-Free** [comment]
I think I understand why some files were missing. I looked into the docker file and it runs a git clone command at the end. Therefore the resulting image contains only files from the original repo and not the files generated by the setup script. I fixed the problem by deleting the git clone command (which was useless at this point as I already had cloned the repo manually) and running the docker image with a shared folder containing all the pyrapose files.
